harness/determined/deploy/healthcheck.py

```diff
@@ -53,3 +53,37 @@ def wait_for_master_url(
     finally:
         if polling:
             print()
+
+
+def wait_for_genai_url(
+    master_url: str,
+    timeout: int = DEFAULT_TIMEOUT,
+    cert: Optional[certs.Cert] = None,
+) -> None:
+    POLL_INTERVAL = 2
+    polling = False
+    start_time = time.time()
+    GENAI_PREFIX = "/genai"
+    check_path = GENAI_PREFIX + "/api/v1/workspaces"
+
+    try:
+        while time.time() - start_time < timeout:
+            try:
+                auth = api.Authentication(master_address=master_url, cert=cert)
```

Comment:
we don't have support for endpoints that do not require authentication at this time.